### PR TITLE
GA Migration Precommit Python Examples ( Python Precommit Sharding)

### PR DESCRIPTION
**Pull request by andoni-guzman**
This PR is part of the migration from jenkins to github actions.
This PR is part of the Precommit Java Sharding

This PR contains this tests

Precommit Python Examples

1 file were added to the workflows folder of github actions

job-precommit-python-examples.yml

CI.md was modified adding the new GA workflows

Waiting for this PR before merge https://github.com/apache/beam/pull/22703

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-precommit-python-examples.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  schedule:
+    - cron: '0 */6 * * *'
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request_target:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/python/apache_beam/examples/**',
+            'sdks/python/apache_beam/examples/**']
```
Comment (Abacn) [inline .github/workflows/job-precommit-python-examples.yml:33]: Is this path duplicated?

**Review by Abacn — COMMENTED** [review]
Thanks. Went through once

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -2611,7 +2611,7 @@ class BeamModulePlugin implements Plugin<Project> {
             def distTarBall = "${pythonRootDir}/build/apache-beam.tar.gz"
             project.exec {
               executable 'sh'
-              args '-c', ". ${project.ext.envdir}/bin/activate && cd ${copiedPyRoot} && scripts/run_tox.sh $tox_env $distTarBall '$posargs'"
+              args '-c', ". ${project.ext.envdir}/bin/activate && cd ${copiedPyRoot} && scripts/run_tox.sh $tox_env $distTarBall $posargs"
```
Comment (damccorm) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2614]: Why did we unquote this?

**Review by damccorm — COMMENTED** [review]
Had one comment, once that is responded to and the duplication Yi called out is addressed this LGTM

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 2 above)
Comment (andoni-guzman) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2614]: There was a parsing error caused by this quotes, in the case that the string contains multiple quotes and double quotes, the solution was deleted this quotes. For example in python examples we had 
`"'apache_beam/examples/complete/game/ \
  apache_beam/examples/complete/autocomplete_test.py \
  apache_beam/examples/complete/distribopt_test.py \`

Gradlew failed to parse this string , failing the execution. 
The reason to execute all the test in one gradle job is to save time and resources. Also I tried using regular expressions but doesn't work , gradle parse it literally. 
Put in a matrix all the jobs generate a lot of executions only for one small test. 
 I tested this change with other workflows and everything was well. But I worried if this change could affect another code or tests.

Comment (damccorm) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:2614]: > I tested this change with other workflows and everything was well. But I worried if this change could affect another code or tests.

That's my concern here too. Can we just drop the `'` for python examples?

Comment (andoni-guzman) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Hi @damccorm  
I did the changes, the quotes get back in the file. I tested again the tests, and everything works well.

**Pull request closed without merging**